src/core_ocean/mode_forward/mpas_ocn_time_integration_split.F

```diff
@@ -559,7 +559,7 @@ subroutine ocn_time_integrator_split(domain, dt)!{{{
 
                allocate(uTemp(nVertLevels))
 
-               !$omp do schedule(runtime) private(cell1, cell2, k, normalThicknessFluxSum, thicknessSum)
+               !$omp do schedule(runtime)
```

Comment:
`private` can be omitted, `firstprivate` is default.

Reply:
@amametjanov could you explain all the changes that remove private(var1, var2,...)?  

I thought we always need to declare indices within an openmp loop to be private, as in the removed line.  It looks like my understand of this is incorrect.  Doesn't each thread need it's own copy of index variables, so the private clause is required?

Reply:
Because of `!$omp parallel default(firstprivate)`, all variables inside `!$omp do` regions are `firstprivate` by default: meaning each thread's copy/storage of a private variable inherits master thread's value. There is no need to make these variables here `private`, because their values are defined/initialized prior to the first use.

Overall, two possible scenarios in an omp-do region: def-before-use, use-before-def. If `firstprivate` variable `var1` is defined before use:
* `private(var1)` is redundant; `var1` is overwritten anyway (this is the reason for removal above)

If `firstprivate` `var1` is used before def:
* `private(var1)` is a bug, intent was to inherit master thread's value

Instead of enumerating all locally defined variables in private clauses, I'd like to simplify and omit them when possible. Otherwise, when a new variable is added to an omp-do region, it also needs to be added to the private clause; when a variable is removed from a region, it also needs to be removed from the private clause. This is hard to do/enforce.

Reply:
Thanks for the explanation.  Let's make sure I understand - is that another way to say it: 
- In MPAS, threads are spawned upon initialization, and run throughout.
- Upon entry to this subroutine, each thread declares its own set of local variables such as `cell1, cell2, k, normalThicknessFluxSum, thicknessSum`, so the memory space is unique to each thread.
- At each loop, `private` is not needed because each thread already has its own copy.

But why do you say "each thread's copy of a private variable inherits master thread's value"? When does that occur - only on initialization?

Reply:
Yes.

The copying occurs just before entering an omp-do loop.